telegramm.py

```diff
@@ -0,0 +1,125 @@
+import getpass
+import time
+
+from selenium import webdriver
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.common.exceptions import TimeoutException, NoSuchElementException
+
+from helpers import number_format
+from database import get_all, insert_into_telegram, is_telegram_acount, create_bd_telegram
+
+create_bd_telegram()
+
+# ToDo Change user_name_pc
+# username = getpass.getuser()
+# print(username)
+
+chrome_options = webdriver.ChromeOptions()
+chrome_options.add_argument("--user-data-dir=C://Users/dkuly/AppData/Local/Google/Chrome/User Data")
+chrome_options.add_experimental_option("useAutomationExtension", False)
+chrome_options.add_argument("--disable-dev-shm-usage")
+chrome_options.add_argument("--disable-gpu")  # applicable to windows os only
+chrome_options.add_argument('--no-sandbox')
+
+driver = webdriver.Chrome(chrome_options=chrome_options)
+driver.maximize_window()
+
+wait = WebDriverWait(driver, 10)
+
+lines = get_all()[:]
+
+driver.get('https://web.telegram.org/#/im')
+time.sleep(20)
+
+for _line in lines:
+    for number in [number_format(item) for item in _line[0].split('\n')]:
+        # ToDo check by Id
+
+        if is_telegram_acount(number):
+            continue
+
+        try:
+            # Addition of phone number
+            print('-' * 50)
+            print(number)
+
+            driver.find_element_by_xpath('//*[@class="tg_head_btn dropdown-toggle"]').click()
+            wait.until(EC.visibility_of_element_located((By.XPATH, '//*[contains(text(),"Contacts")]'))).click()
+            driver.find_element_by_xpath('//*[contains(text(),"New contact")]').click()
+
+            actions = ActionChains(driver)
+            actions.move_to_element(driver.find_element_by_xpath('//*[contains(text(),"First name")]'))
+            actions.click()
+            actions.move_to_element(driver.find_element_by_xpath('//*[contains(text(),"Phone number")]'))
+            actions.click()
+            actions.send_keys(number)
+            actions.perform()
+            actions.reset_actions()
+
+            driver.find_element_by_xpath('//*[contains(text(),"Save")]').click()
+
+            # Error Verification
+            try:
+                wait_err = WebDriverWait(driver, 3)
```

Comment:
Is using of implicit waits really necessary when you have an explicit waiter right after it?

Reply:
have discussed